insights/parsers/tomcat_virtual_dir_context.py

```diff
@@ -0,0 +1,46 @@
+"""
+TomcatVirtualDirContext - command ``/bin/grep -R --include '*.xml' 'VirtualDirContext' /usr/share/tomcat*``
+===========================================================================================================
+
+This parser reads the output of ``/bin/grep -R --include '*.xml' 'VirtualDirContext' /usr/share/tomcat*``
+and detects whether there are any config files which contain VirtualDirContext.
+
+Sample input::
+
+    /usr/share/tomcat/conf/server.xml:    <Resources className="org.apache.naming.resources.VirtualDirContext"
+
+Examples::
+
+    >>> shared[TomcatVirtualDirContext].has_virtual_dir_context
+    True
+    >>> shared[TomcatVirtualDirContext].lines
+    ('/usr/share/tomcat/conf/server.xml',
+    '    <Resources className="org.apache.naming.resources.VirtualDirContext"')
+"""
+
+from .. import Parser, parser
+from . import ParseException
+
+
+@parser('tomcat_virtual_dir_context')
+class TomcatVirtualDirContext(Parser):
+    """
+    Reports whether there is a VirtualDirContext option used in TomCat configuration file.
+    """
+    def __init__(self, *args, **kwargs):
+        self.has_virtual_dir_context = False
+        """bool: True if VirtualDirContext is used."""
+        self.lines = []
```

Comment:
It would be convenient to store/parse the `file_name` and `file_line` as a `dict` than `list`.
~~~
self.data = {'/usr/share/tomcat/conf/server.xml': 
                   ['<Resources className="org.apache.naming.resources.VirtualDirContext'],  <-- if line is unique in one .xml file, `str` is more suitable here than `list`
}
~~~

And add several useful interfaces like `files_has_virtual_dir_context`:
~~~
def files_has_virtual_dir_context(self):
    return self.data.keys()
~~~

Reply:
Storing in a dictionary in case that there are more lines with VirtualDirContext in one file makes sense. Will be updated. 

However, I do not need any extra interfaces for the rule, as I will just iterate through the dictionary and display offending files / lines. Also, there is a very high probability that this parser will never be re-used.